### PR TITLE
K8SPXC-391 check if pmm container should be added

### PR DESCRIPTION
**Pull request by AlyHKafoury**
[K8SPXC-391](https://jira.percona.com/browse/K8SPXC-391)

### The behavior before this PR:
when PMM is enabled, `PMMContainer` functions returns empty `corev1.Container` structure, which becomes added into HAProxy Pod definition.
As a result, HAProxy pod cannot be started because of error `[spec.containers[0].name: Required value, spec.containers[0].image: Required value]`

### The behavior after this PR:
If PMM is enabled, `PMMContainer` functions returns `nil` value.
Code checks returned value and add value only if it is not `nil`.
As a result, HAProxy pod doesn't have a PMM Container. So HAProxy Pod can run without any issues.

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| scaling | passed |
| recreate | passed |
| big-data | passed |
| init-deploy | passed |
| self-healing | passed |
| scaling-proxysql | passed |
| limits | passed |
| monitoring | failed |
| restore-to-encrypted-cluster | failed |
| upgrade | passed |
| demand-backup | passed |
| self-healing-advanced | failed |
| upgrade-consistency | failed |
| operator-self-healing | passed |
| security-context | passed |
| monitoring-2-0 | failed |
| scheduled-backup | failed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| users | passed |
| demand-backup-encrypted-with-tls | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| scaling | passed |
| storage | passed |
| recreate | passed |
| big-data | passed |
| self-healing | passed |
| limits | passed |
| scaling-proxysql | passed |
| restore-to-encrypted-cluster | failed |
| upgrade | passed |
| self-healing-advanced | passed |
| demand-backup | passed |
| monitoring | passed |
| operator-self-healing | passed |
| upgrade-consistency | failed |
| security-context | passed |
| monitoring-2-0 | passed |
| scheduled-backup | failed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| demand-backup-encrypted-with-tls | failed |
| proxysql-sidecar-res-limits | passed |
| users | passed |

**Comment by blez** [comment]
Could you please add some context here? Because HAProxy was mentioned in title of PR, but I can't see ant code changes for HAProxy - only for PMM.

**Comment by AlyHKafoury** [comment]
> Could you please add some context here? Because HAProxy was mentioned in title of PR, but I can't see any code changes for HAProxy - only for PMM.

I disabled pmm for HAProxy as it is not required for it that's all @blez

File: pkg/controller/pxc/upgrade.go [hunk 1]
```diff
@@ -79,8 +79,9 @@ func (r *ReconcilePerconaXtraDBCluster) updatePod(sfs api.StatefulApp, podSpec *
 		if err != nil {
 			return fmt.Errorf("pmm container error: %v", err)
 		}
-
-		newContainers = append(newContainers, pmmC)
+		if pmmC.Image != "" {
```
Comment (delgod) [inline pkg/controller/pxc/upgrade.go:82]: I belive we must check if PMM is enabled in general
```suggestion
		if cr.Spec.PMM != nil && cr.Spec.PMM.Enabled {
```

Comment (AlyHKafoury) [reply pkg/controller/pxc/upgrade.go:82]: @delgod pmm is enabled or not is not the case here as we don't need for haproxy in either way as it is enabled for the other parts like pxc or proxysql

Comment (delgod) [reply pkg/controller/pxc/upgrade.go]: Image can be passed from CR or from Version Service, it is not reliable, I have reworked on `pmmC.Name`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| scaling | passed |
| recreate | passed |
| init-deploy | passed |
| big-data | passed |
| self-healing | passed |
| scaling-proxysql | passed |
| limits | passed |
| restore-to-encrypted-cluster | passed |
| monitoring | passed |
| upgrade | passed |
| self-healing-advanced | passed |
| monitoring-2-0 | passed |
| demand-backup | passed |
| upgrade-consistency | passed |
| affinity | passed |
| security-context | passed |
| operator-self-healing | passed |
| one-pod | passed |
| auto-tuning | passed |
| scheduled-backup | failed |
| proxysql-sidecar-res-limits | passed |
| users | passed |
| demand-backup-encrypted-with-tls | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| big-data | passed |
| storage | passed |
| init-deploy | passed |
| recreate | passed |
| scaling | passed |
| self-healing | passed |
| scaling-proxysql | passed |
| restore-to-encrypted-cluster | passed |
| upgrade | passed |
| self-healing-advanced | passed |
| limits | passed |
| upgrade-consistency | passed |
| demand-backup | passed |
| security-context | passed |
| monitoring | passed |
| monitoring-2-0 | passed |
| operator-self-healing | passed |
| affinity | passed |
| scheduled-backup | failed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| users | passed |
| demand-backup-encrypted-with-tls | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| big-data | passed |
| storage | passed |
| recreate | passed |
| init-deploy | passed |
| scaling | passed |
| self-healing | passed |
| scaling-proxysql | passed |
| restore-to-encrypted-cluster | passed |
| upgrade | passed |
| self-healing-advanced | passed |
| upgrade-consistency | passed |
| limits | passed |
| monitoring | passed |
| security-context | passed |
| operator-self-healing | passed |
| demand-backup | passed |
| monitoring-2-0 | passed |
| affinity | passed |
| scheduled-backup | failed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| users | passed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| storage | passed |
| scaling | passed |
| recreate | passed |
| big-data | passed |
| self-healing | passed |
| scaling-proxysql | passed |
| limits | passed |
| monitoring | passed |
| restore-to-encrypted-cluster | passed |
| upgrade | passed |
| self-healing-advanced | passed |
| monitoring-2-0 | failed |
| demand-backup | passed |
| operator-self-healing | passed |
| upgrade-consistency | passed |
| affinity | passed |
| security-context | passed |
| scheduled-backup | failed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| users | passed |
| demand-backup-encrypted-with-tls | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| scaling | passed |
| recreate | passed |
| init-deploy | passed |
| big-data | passed |
| limits | passed |
| scaling-proxysql | passed |
| self-healing | passed |
| monitoring | passed |
| restore-to-encrypted-cluster | passed |
| upgrade | passed |
| monitoring-2-0 | failed |
| self-healing-advanced | passed |
| upgrade-consistency | passed |
| demand-backup | passed |
| security-context | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| operator-self-healing | passed |
| scheduled-backup | failed |
| proxysql-sidecar-res-limits | passed |
| users | passed |
| demand-backup-encrypted-with-tls | failed |

File: e2e-tests/monitoring-2-0/compare/statefulset_monitoring-haproxy.yml [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        volumeMounts:
+        - mountPath: /etc/haproxy-custom/
+          name: haproxy-custom
+        - mountPath: /etc/haproxy/pxc
+          name: haproxy-auto
+      dnsPolicy: ClusterFirst
+      restartPolicy: Always
+      schedulerName: default-scheduler
+      securityContext: {}
+      serviceAccount: percona-xtradb-cluster-operator-workload
```
Comment (hors) [inline e2e-tests/monitoring-2-0/compare/statefulset_monitoring-haproxy.yml:91]:
```suggestion
      serviceAccount: default
```

File: e2e-tests/monitoring-2-0/compare/statefulset_monitoring-haproxy.yml [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+      serviceAccountName: percona-xtradb-cluster-operator-workload
```
Comment (hors) [inline e2e-tests/monitoring-2-0/compare/statefulset_monitoring-haproxy.yml:92]:
```suggestion
      serviceAccountName: default
```

**Review by hors — CHANGES_REQUESTED** [review]

**Review by delgod — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scaling | passed |
| recreate | passed |
| scaling-proxysql | passed |
| restore-to-encrypted-cluster | passed |
| upgrade | passed |
| demand-backup | passed |
| upgrade-consistency | passed |
| scheduled-backup | failed |
| security-context | passed |
| big-data | passed |
| init-deploy | passed |
| limits | passed |
| monitoring | passed |
| monitoring-2-0 | failed |
| storage | passed |
| self-healing | passed |
| self-healing-advanced | passed |
| operator-self-healing | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| users | passed |
| demand-backup-encrypted-with-tls | failed |

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scaling | passed |
| big-data | passed |
| storage | passed |
| recreate | passed |
| init-deploy | passed |
| scaling-proxysql | passed |
| self-healing | passed |
| limits | passed |
| upgrade | passed |
| restore-to-encrypted-cluster | passed |
| monitoring | passed |
| upgrade-consistency | passed |
| self-healing-advanced | passed |
| security-context | passed |
| monitoring-2-0 | passed |
| operator-self-healing | passed |
| demand-backup | passed |
| affinity | passed |
| one-pod | passed |
| scheduled-backup | failed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| users | passed |
| demand-backup-encrypted-with-tls | failed |

**Review by hors — APPROVED** [review]

**Pull request merged**